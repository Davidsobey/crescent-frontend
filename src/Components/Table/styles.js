import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Actions = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const QuotesLink = styled(Link)`
  background-color: #ffdd47;
  padding: 5px 10px;
  border-radius: 50px;
`;

export const TableWrapper = styled.div``;

export const FiltersWrapper = styled.div`
  display: flex;
  align-items: center;
  padding-bottom: 15px;
`;

export const FilterItem = styled.div`
  margin-right: 15px;
`;
