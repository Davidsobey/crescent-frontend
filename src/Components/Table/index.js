import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Table, {
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from "material-ui/Table";
import Paper from "material-ui/Paper";

const styles = theme => ({
  root: {
    width: "100%",
    marginTop: theme.spacing.unit * 3,
    overflowX: "auto"
  },
  table: {
    minWidth: 700
  }
});

function loadRows(obj, arr) {
  for (const prop in obj) {
    if (prop !== "id") {
      const value = obj[prop];
      arr.push(value);
    }
  }
  return arr;
}

function BasicTable(props) {
  const { classes, data, header } = props;
  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            {header.map(head => {
              return <TableCell key={head}>{head}</TableCell>;
            })}
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map(obj => {
            const arr = [];
            loadRows(obj, arr);
            return (
              <TableRow key={obj.id}>
                {arr.map(value => <TableCell key={value}>{value}</TableCell>)}
              </TableRow>
            );
          })}
        </TableBody>
      </Table>
    </Paper>
  );
}

BasicTable.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(BasicTable);
