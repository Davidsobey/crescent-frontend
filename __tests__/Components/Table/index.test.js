import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import Table from '../../../src/Components/Table/index.js';


const mockStore = configureStore();
const initialState = {};
const store = mockStore(initialState);

const header = ['Name', 'View Material'];
const data = [{'name':'aaa', 'button':''}];


describe('<Table />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Table store={store} header={header} data={data} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});