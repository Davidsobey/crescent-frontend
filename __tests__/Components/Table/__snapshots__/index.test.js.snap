// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> render() renders the component 1`] = `
<WithStyles(Paper)>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          key="Name"
        >
          Name
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          key="View Material"
        >
          View Material
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          key="1"
        >
          aaa
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</WithStyles(Paper)>
`;
